### PR TITLE
feat: migrate web3 client to ethers.rs

### PR DESCRIPTION
**Pull request by atanmarko**
# Description

Migrate `web3` sequencer subnet client to `ethers` library.

Fixes TP-516

## PR Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/topos-sequencer-subnet-client/Cargo.toml [hunk 1]
```diff
@@ -8,12 +8,15 @@ hex.workspace = true
...
 tiny-keccak.workspace = true
+ethers.workspace = true
+ethers-providers = { version = "2.0.4", features = ["ws"] }
+rustc-hex = "2.1.0"
 
-topos-core = { workspace = true, features = ["uci"] }
+topos-core = { workspace = true, features = ["uci", "api"] }
+
+[build-dependencies]
+ethers.workspace = true
```
Comment (Freyskeyd) [inline crates/topos-sequencer-subnet-client/Cargo.toml:22]: need a blank line at the end

File: crates/topos-tce/Cargo.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name = "topos-tce"
 description = "TCE Node Service"
 version = "0.1.0"
 edition = "2021"
-rust-version = "1.64"
+rust-version = "1.65"
```
Comment (Freyskeyd) [inline crates/topos-tce/Cargo.toml:6]: did you use MSRV to check for compatibility?

File: crates/topos-sequencer-subnet-runtime/src/proxy.rs [hunk 3]
```diff
@@ -270,14 +270,16 @@ impl SubnetRuntimeProxy {
...
+    ) -> Result<Option<String>, Error> {
         debug!(
             "Pushing certificate with id {} to target subnet {}, tcc {}",
             cert.id, runtime_proxy_config.subnet_id, runtime_proxy_config.subnet_contract_address,
         );
         let receipt = subnet_client.push_certificate(cert, position).await?;
         debug!("Push certificate transaction receipt: {:?}", &receipt);
-        Ok("0x".to_string() + &hex::encode(receipt.transaction_hash))
+        let tx_hash =
+            receipt.map(|tx_receipt| "0x".to_string() + &hex::encode(tx_receipt.transaction_hash));
```
Comment (Freyskeyd) [inline crates/topos-sequencer-subnet-runtime/src/proxy.rs:281]: Could we move this to a `Display` impl somewhere?

**Review by Freyskeyd — APPROVED** [review]

File: crates/topos-sequencer-subnet-runtime/src/proxy.rs (hunk 3 above)
Comment (atanmarko) [reply crates/topos-sequencer-subnet-runtime/src/proxy.rs:281]: Where do you suggest we should move it? `receipt` It is an `ethers` structure.

File: crates/topos-tce/Cargo.toml (hunk 2 above)
Comment (atanmarko) [reply crates/topos-tce/Cargo.toml:6]: Yes it said:
```
Finished The MSRV is: 1.65.0 
```

**Pull request merged**